File: backend/src/main/java/hu/unideb/inf/NeptunPro/model/student/Student.java
package hu.unideb.inf.NeptunPro.model.student;

import hu.unideb.inf.NeptunPro.model.Program;
import hu.unideb.inf.NeptunPro.model.user.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Student {

    @Id
    @GeneratedValue
    private Long id;

    private String firstname;

    private String lastname;

    private Long programId;

    private Double gpa;

    private Integer version;

    private Long createdBy;
    private Long modifiedBy;

    @CreationTimestamp
    private Timestamp createdAt;

    @UpdateTimestamp
    private Timestamp modifiedAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "programId", referencedColumnName = "id", insertable = false, updatable = false)
    private Program programInfo;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "createdBy", referencedColumnName = "id", insertable = false, updatable = false)
    private User createdByInfo;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "modifiedBy", referencedColumnName = "id", insertable = false, updatable = false)
    private User modifiedByInfo;

}

File: frontend/neptunpro/lib/data/models/student.dart
import 'package:neptunpro/data/models/modificator_info.dart';

class Student {
  int id;
  String firstname;
  String lastname;
  int programId;
  double gpa;
  int version;
  int createdBy;
  int modifiedBy;
  DateTime createdAt;
  DateTime modifiedAt;
  ProgramInfo programInfo;
  ModificatorInfo createdByInfo;
  ModificatorInfo modifiedByInfo;

  Student({
    required this.id,
    required this.firstname,
    required this.lastname,
    required this.programId,
    required this.gpa,
    required this.version,
    required this.createdBy,
    required this.modifiedBy,
    required this.createdAt,
    required this.modifiedAt,
    required this.programInfo,
    required this.createdByInfo,
    required this.modifiedByInfo,
  });

  factory Student.fromJson(Map<String, dynamic> map) {
    return Student(
      id: map['id'] as int,
      firstname: map['firstname'] as String,
      lastname: map['lastname'] as String,
      programId: map['programId'] as int,
      gpa: map['gpa'] as double,
      version: map['version'] as int,
      createdBy: map['createdBy'] as int,
      modifiedBy: map['modifiedBy'] as int,
      createdAt: DateTime.parse(map['createdAt']),
      modifiedAt: DateTime.parse(map['modifiedAt']),
      programInfo: ProgramInfo.fromJson(map['programInfo'] as Map<String,dynamic>),
      createdByInfo: ModificatorInfo.fromJson(map['createdByInfo'] as Map<String,dynamic>),
      modifiedByInfo: ModificatorInfo.fromJson(map['modifiedByInfo'] as Map<String,dynamic>),
    );
  }
}

class ProgramInfo {
  int id;
  String name;
  ProgramInfo({
    required this.id,
    required this.name,
  });

  factory ProgramInfo.fromJson(Map<String, dynamic> map) {
    return ProgramInfo(
      id: map['id'] as int,
      name: map['name'] as String,
    );
  }

}
